cluster_agent/settings.py

```diff
@@ -54,6 +52,20 @@ class Settings(BaseSettings):
     # Single user submitter settings
     SINGLE_USER_SUBMITTER: Optional[str]
 
+    @validator("BASE_SLURMRESTD_URL", always=True)
+    def _validate_slurm_version_on_url(cls, v):
+        """
+        Add the default slurm version to the url if it is not present, for backward compatibility.
+        """
+        if not re.match(r".*/slurm/v\d+\.\d+\.\d+$", v):
+            default_version = "/slurm/v0.0.36"
+            logger.warning(
+                "Slurmrestd version not found in the URL, defaulting to {}",
+                default_version,
+            )
+            v += default_version
```

Comment:
I feel like this could be error prone. For example, having `BASE_SLURMRESTD_URL=/slurm/v0.36` would fail the regex above, but the resulting URL would be `/slurm/v0.36/slurm/v0.0.36".

I think it would be easier to have a SLURM_RESTD_VERSION to build the URL and just flat error out when the version scheme doesn't match the regex.

Reply:
That is a good point, this logical comparison can be refined. The goal here is not to check the syntax but to add a default version if it is not there, a temporary approach for backward compatibility.

Reply:
removed on [d5bc04a](https://github.com/omnivector-solutions/cluster-agent/pull/72/commits/d5bc04abab51bd04a63b624dcec28c1ced303b3d)